### PR TITLE
add duration to iOS SDK for Chosen

### PR DESCRIPTION
**Pull request by aaustin**
@qinweigong urgent

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BNCLinkData.h [hunk 1]
```diff
@@ -20,6 +20,7 @@
 @property (nonatomic, strong) NSString *feature;
 @property (nonatomic, strong) NSString *stage;
 @property (nonatomic, strong) NSString *params;
+@property (nonatomic, assign) NSInteger duration;
```
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/BNCLinkData.h]: To make it clear to partners, would it be better to declare it as NSUInteger?

**Comment by aaustin** [comment]
Changed to NSU and realized I forgot update the link data @qinweigong

**Comment by qinweigong** [comment]
:+1:  Could you also change it in BNCLinkData.h property declaration?

File: Branch-SDK/Branch-SDK/Branch.m [hunk 2]
```diff
@@ -606,108 +606,125 @@ - (NSDictionary *)getLatestReferringParams {
...
+    [self generateShortUrl:tags andAlias:nil andType:BranchLinkTypeUnlimitedUse andMatchDuration:0 andChannel:channel andFeature:BRANCH_FEATURE_TAG_SHARE andStage:nil andParams:[BranchServerInterface encodePostToUniversalString:[self sanitizeQuotesFromInput:params]] andCallback:callback];
 }
 
 - (void)getContentUrlWithParams:(NSDictionary *)params andChannel:(NSString *)channel andCallback:(callbackWithUrl)callback {
-    [self generateShortUrl:nil andAlias:nil andType:BranchLinkTypeUnlimitedUse andChannel:channel andFeature:BRANCH_FEATURE_TAG_SHARE andStage:nil andParams:[BranchServerInterface encodePostToUniversalString:[self sanitizeQuotesFromInput:params]] andCallback:callback];
+    [self generateShortUrl:nil andAlias:nil andType:BranchLinkTypeUnlimitedUse andMatchDuration:0 andChannel:channel andFeature:BRANCH_FEATURE_TAG_SHARE andStage:nil andParams:[BranchServerInterface encodePostToUniversalString:[self sanitizeQuotesFromInput:params]] andCallback:callback];
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: Is defaulting this to 0 the right thing to do?

Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: I guess in that case, it will just be ignored. Do we want to have a default match duration that is non-zero?

**Comment by shortstuffsushi** [comment]
Is there any explanation as to what this "duration" parameter is, or what it's used for? Should that be added to the Readme or anything?

File: Branch-SDK/Branch-SDK/Branch.m (hunk 2 above)
Comment (aaustin) [inline Branch-SDK/Branch-SDK/Branch.m]: It's a default set on the backend - there's no need to send/hardcode it client side

**Comment by aaustin** [comment]
When someone clicks a link, there is an outstanding click object created that is eligible for match when an app opens up. This duration is the amount of time that the click object is eligible for match

**Comment by shortstuffsushi** [comment]
So, this would be relevant when you're *creating* a link, allowing you to specify how long that link, when clicked, should keep an open match. Gotcha.

**Pull request merged**